src/routes/add-article.ts

```diff
@@ -1,27 +1,33 @@
+import clownface from 'clownface';
 import createHttpError from 'http-errors';
 import { constants } from 'http2';
-import jsonld from 'jsonld';
-import { JsonLdArray } from 'jsonld/jsonld-spec';
 import { Next } from 'koa';
-import { schema } from 'rdf-namespaces';
-import uniqueString from 'unique-string';
 import { AppContext, AppMiddleware } from '../app';
 import NotAnArticle from '../errors/not-an-article';
+import { rdf, schema } from '../namespaces';
 
 export default (): AppMiddleware => (
   async ({ articles, request, response }: AppContext, next: Next): Promise<void> => {
-    const [article] = await jsonld.expand(request.body) as JsonLdArray;
-    if ('@id' in article) {
-      throw new createHttpError.Forbidden(`Article IDs must not be set ('${article['@id']}' was given)`);
+    const graph = clownface({ dataset: request.dataset });
+
+    const foundArticles = graph.has(rdf.type, schema.Article);
+
+    if (foundArticles.terms.length > 1) {
+      throw new createHttpError.BadRequest('Multiple articles found');
     }
-    if (!(schema.name in article) || article[schema.name].length === 0) {
-      throw new createHttpError.BadRequest(`Article must have at least one ${schema.name}`);
+
+    if (foundArticles.terms.length === 0) {
+      throw new createHttpError.BadRequest(`No ${schema.Article.value} found`);
     }
 
-    article['@id'] = `_:${uniqueString()}`;
+    const id = foundArticles.terms[0];
```

Comment:
Need a check to make sure this a blank node, then migrate all the statements to a new unique blank node (will be a named node in the future).